### PR TITLE
fix: Re-apply tags in editor when project tags are updated

### PR DESCRIPTION
**Pull request by wbreza**
Resolves issue where if tag color is modified while in the editor page, the color changes are not applied till you reload or navigate away from the current asset

Resolves AB#17264

### CODE REVIEW

File: src/react/components/pages/editorPage/canvas.test.tsx [hunk 1]
```diff
@@ -568,11 +568,26 @@ describe("Editor Canvas", () => {
...
+        const project = { ...wrapper.props().project };
+        const tags = [...project.tags];
+        tags[0].color = "#FFCCFF";
+        project.tags = tags;
+
+        const editor = wrapper.instance().editor;
+        const updateTagsSpy = jest.spyOn(editor.RM, "updateTagsById");
+
+        wrapper.setProps({ project });
+        expect(updateTagsSpy).toBeCalledTimes(assetMetadata.regions.length);
```
Comment (mydiemho) [inline src/react/components/pages/editorPage/canvas.test.tsx]: I feel like this is exposing/dependent on internal implementation.  How about testing that all the regions have a new tag color?

File: src/react/components/pages/editorPage/canvas.test.tsx [hunk 2]
```diff
@@ -512,7 +512,7 @@ describe("Editor Canvas", () => {
                 ...cProps.selectedAsset,
                 regions: [copiedRegion],
             },
-        }).find(Canvas);
```
Comment (mydiemho) [inline src/react/components/pages/editorPage/canvas.test.tsx]: how does the tests still pass with this gone / how does this not break tests before?

Comment (wbreza) [reply src/react/components/pages/editorPage/canvas.test.tsx]: The `wrapper` is the canvas and this isn't needed.  If you take a look at `createComponent()` the root element that it returns is the `Canvas` component itself.

File: src/react/components/pages/editorPage/canvas.test.tsx (hunk 1 above)
Comment (wbreza) [reply src/react/components/pages/editorPage/canvas.test.tsx]: Our tag color is modified directly by the tags input.  This bug that the canvas component didn't update to reflect tag updates.  Extending this test beyond calling into canvas tools API would be re-testing the internal implementation of the canvas itself.

**Pull request merged**